temporalio/client.py

```diff
@@ -2821,6 +2915,11 @@ def __init__(
             self.search_attributes = temporalio.converter.decode_search_attributes(
                 raw_info.search_attributes
             )
+            self.typed_search_attributes = (
```

Comment:
How do you handle updating a schedule action?  If I try to do an update on a schedule action  that has a search attribute 
 set won't I have `search_attributes` and `typed_search_attributes` and your validation in `_encode_maybe_typed_search_attributes` will fail?

Reply:
Let me confirm and get back. This might be a hole in my logic.

Reply:
Ok, this is indeed a hole. There are a few options here but they are all a bit gross. The best way probably is, since both old form and new form are immutable, to set a "__from_server: True" Python attribute on them and, when serializing for update specifically, use untyped when both have that set, or use the one without that attribute set if only one has it set, or fail if they both don't have that attribute set.

Can't cheat like this in languages like Go though.

I'm open to better options with the constraint that we can't break what's there today.

Reply:
Fixed by replacing `search_attributes` with `typed_search_attributes`